app/models/project_step.rb

```diff
@@ -123,4 +123,46 @@ def scheduled_bg
       self.background_color
     end
   end
+
+  def scheduled_day
+    self.scheduled_date.day
+  end
+
+  def scheduled_weekday
+    Date::DAYNAMES[self.scheduled_date.wday]
+  end
+
+  def scheduled_weekday_key
+    self.nth_weekday.to_s + "_" + self.scheduled_weekday.downcase
+  end
+
+  def nth_weekday
```

Comment:
I'm struggling to think of a better name for this, but I found myself a little confused by what it was supposed to be doing judging based just on the name. I wonder whether we should try to come up with something a little more descriptive.